chia/types/mojos.py

```diff
@@ -0,0 +1,29 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from decimal import Decimal
+from typing import final, Type, TypeVar
+
+from chia.cmds.units import units
+from chia.util.ints import uint64
+
+_T_Mojos = TypeVar("_T_Mojos", bound="Mojos")
+
+
+@dataclass
+@final
+class Mojos:
+    """
+    XCH conversion: 1 Chia = 1 Trillion Mojos
+    """
+
+    mojos: uint64
+
+    def __init__(self, mojos: uint64):
+        self.mojos = mojos
+
+    @classmethod
+    def from_chia(cls: Type[_T_Mojos], chia_amount: int) -> Mojos:
```

Comment:
It was more of an either/or situation.  Since you have `@final` and `-> Mojos`, the hint on `cls` can be dropped.

```suggestion
    def from_chia(cls, chia_amount: int) -> Mojos:
```

Reply:
Ya, I know